app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -226,7 +233,87 @@ enum SectionHeaderPlacement {
 const SECTION_HEADER_PLACEMENT: SectionHeaderPlacement = SectionHeaderPlacement::Bottom;
 
 
+
+// ========================
 // === Shape Definition ===
+// ========================
+
+
+// === Selection ===
+
+/// A shape of a selection box. It is used as a mask to show only specific parts of the selection
+/// layers.
+///
+/// The selection box highlights a selected entry in the component list panel view. We render the
+/// selection above the background of the component group (so it is visible) but below the entry's
+/// text (so the text is not covered). Headers of the component groups complicate matters. The
+/// header is displayed on top of entries when scrolling the group, so there are cases when
+/// selection covers the header background (the header is considered the entry) and cases when the
+/// header covers selection, i.e., we select a partially visible entry below the header. To allow
+/// such behavior, we render a highlighted version of each component group on top of a normal one
+/// and use a layer masking to show only a part of it. See
+/// [component_group](component_group#selection) documentation to learn more about how the component
+/// groups are implemented.
+///
+/// The selection box should never leave the borders of the component list panel. We can't use
+/// [layers masking][] as our renderer does not support hierarchical masks, so instead, we build a
+/// final shape as an intersection of the `mask` rounded rectangle and the `area` shape. The `area`
+/// covers the entire component list panel and thus limits the area in which the selection box is
+/// visible.
+///
+/// The shape of the component list panel determines the size and the position of the `area`. One
+/// should use the display object's standard `set_position` API to move the `area`. The `mask` moves
+/// freely and represents the position of the selection box itself. There are `pos` and
+/// `selection_size` parameters for the position and the size of the `mask`, respectively.
+///
+/// The `margin_top` parameter controls the variable height of the `area`. In a scrolled
+/// component group, the user can select a partially visible entry behind the group's header. We
+/// make the selection box appear covered by the selection box by reducing the height of the `area.`
+/// It works because the header is always on top of the component list panel in a scrolled group.
```

Comment:
I don't understand the sentence. Why selection box masked by selection box always work because header is on top?

Reply:
That was a typo, thanks for catching it.